src/libraries/Microsoft.Extensions.Logging/src/LoggerInformation.cs

```diff
@@ -18,13 +18,13 @@ public MessageLogger(ILogger logger, string category, string providerTypeFullNam
 
         public ILogger Logger { get; }
 
-        public string Category { get; }
+        public string? Category { get; }
 
-        private string ProviderTypeFullName { get; }
+        private string? ProviderTypeFullName { get; }
 
         public LogLevel? MinLevel { get; }
 
-        public Func<string, string, LogLevel, bool> Filter { get; }
+        public Func<string?, string?, LogLevel?, bool>? Filter { get; }
```

Comment:
```suggestion
        public Func<string?, string?, LogLevel, bool>? Filter { get; }
```

Making this change, and then flowing it all the way through the code, should fix a lot of your problems.